### PR TITLE
s3fs: fix dependencies

### PR DESCRIPTION
**Pull request by teh**
###### Motivation for this change

python3Modules.s3fs is currently broken due to an upstream dependency change (e.g. https://hydra.nixos.org/job/nixpkgs/trunk/python37Packages.s3fs.x86_64-darwin).

This PR inits the missing dependencies at their most recent releases and fixes the s3fs build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/aiobotocore/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  # relax version constraints: aiobotocore works with newer botocore versions
+  # the pinning used to match some `extras_require` we're not using.
+  preConfigure = ''
+    substituteInPlace setup.py --replace 'botocore>=1.17.44,<1.17.45' 'botocore'
+  '';
+
+  propagatedBuildInputs = [ wrapt aiohttp aioitertools botocore ];
+
+  # tests not distributed on pypi
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aiobotocore/default.nix:33]:
```suggestion
  doCheck = false;
  
  pythonImportsCheck = [ "aiobotocore" ];
```

File: pkgs/development/python-modules/aioitertools/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18ql6k2j1839jf2rmmmm29v6fb7mr59l75z8nlf0sadmydy6r9al";
+  };
+
+  propagatedBuildInputs = [ typing-extensions ];
+  checkInputs = [ coverage toml mypy ];
+
+  checkPhase = ''
+    make test
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aioitertools/default.nix:26]: We only want to execute this command https://github.com/omnilib/aioitertools/blob/main/makefile#L30.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aioitertools/default.nix (hunk 2 above)
Comment (teh) [reply pkgs/development/python-modules/aioitertools/default.nix]: done

File: pkgs/development/python-modules/aiobotocore/default.nix (hunk 1 above)
Comment (teh) [reply pkgs/development/python-modules/aiobotocore/default.nix:33]: done

**Review by teh — COMMENTED** [review]
thanks!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108059` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.aiobotocore</li>
<li>python37Packages.aioitertools</li>
<li>python37Packages.s3fs</li>
<li>python38Packages.aiobotocore</li>
<li>python38Packages.aioitertools</li>
<li>python38Packages.s3fs</li>
<li>python39Packages.aiobotocore</li>
<li>python39Packages.aioitertools</li>
<li>python39Packages.s3fs</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108059` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.aiobotocore</li>
    <li>python37Packages.aioitertools</li>
    <li>python37Packages.s3fs</li>
    <li>python38Packages.aiobotocore</li>
    <li>python38Packages.aioitertools</li>
    <li>python38Packages.s3fs</li>
    <li>python39Packages.aiobotocore</li>
    <li>python39Packages.aioitertools</li>
    <li>python39Packages.s3fs</li>
  </ul>
</details>

File: pkgs/development/python-modules/aioitertools/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18ql6k2j1839jf2rmmmm29v6fb7mr59l75z8nlf0sadmydy6r9al";
+  };
+
+  propagatedBuildInputs = [ typing-extensions ];
+  checkInputs = [ coverage toml ];
+
+  checkPhase = ''
+    python -m coverage run -m aioitertools.tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aioitertools/default.nix:25]:
```suggestion
    ${python.interpreter} -m coverage run -m aioitertools.tests
```

File: pkgs/development/python-modules/aioitertools/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, typing-extensions
+, coverage
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aioitertools/default.nix:7]:
```suggestion
, python
, coverage
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108059` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.aiobotocore</li>
<li>python37Packages.aioitertools</li>
<li>python37Packages.s3fs</li>
<li>python38Packages.aiobotocore</li>
<li>python38Packages.aioitertools</li>
<li>python38Packages.s3fs</li>
<li>python39Packages.aiobotocore</li>
<li>python39Packages.aioitertools</li>
<li>python39Packages.s3fs</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108059` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.aiobotocore</li>
    <li>python37Packages.aioitertools</li>
    <li>python37Packages.s3fs</li>
    <li>python38Packages.aiobotocore</li>
    <li>python38Packages.aioitertools</li>
    <li>python38Packages.s3fs</li>
    <li>python39Packages.aiobotocore</li>
    <li>python39Packages.aioitertools</li>
    <li>python39Packages.s3fs</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108059` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.aiobotocore</li>
<li>python37Packages.aioitertools</li>
<li>python37Packages.s3fs</li>
<li>python38Packages.aiobotocore</li>
<li>python38Packages.aioitertools</li>
<li>python38Packages.s3fs</li>
<li>python39Packages.aiobotocore</li>
<li>python39Packages.aioitertools</li>
<li>python39Packages.s3fs</li>
</ul>
</details>

**Pull request merged**